cfme/services/catalogs/orchestration_template.py

```diff
@@ -45,8 +64,34 @@ def _orch_templates_create_dialog(context):
 class OrchestrationTemplate(Updateable, Pretty):
     pretty_attrs = ['template_type', 'template_name']
 
-    def __init__(self, template_type=None):
+    def __init__(self, template_type=None, name=None, description=None):
         self.template_type = template_type
+        self.name = name
+        self.description = description
+
+    def create(self, content):
+        sel.force_navigate('create_new_template',
+                           context={'template_type': self.template_type})
+        if(self.template_type == "CloudFormation Templates"):
+            temp_type = "Amazon CloudFormation"
+        else:
+            temp_type = "OpenStack Heat"
+        fill(create_template_form, {'name': self.name,
+                                    'description': self.description,
+                                    'template_type': temp_type,
+                                    'content': content},
+            action=create_template_form.add_button)
+        flash.assert_success_message('Orchestration Template "%s" was saved' %
+                                     self.name)
+
+    def delete(self, template_type, template_name):
+        sel.force_navigate('select_template',
+                           context={'template_type': self.template_type,
+                                    'template_name': template_name})
+        cfg_btn("Remove this Orchestration Template")
```

Comment:
Maybe try `cfg_btn("Remove this Orchestration Template", invokes_alert=True)` here instead of manually handling the assert?